### DISCUSSION TITLE
Random Klipper Shutdown - MCU Connection Issue - SV06

### DISCUSSION THREAD

**Discussion by klb331** [bassamanator/Sovol-SV06-firmware#189] [Q&A]
I’m about done with Klipper. 

For weeks I’ve been working on configuration, tuning, updating and spending and wasting filament getting Klipper to work and the damn thing crashes randomly with no explanation and the log files suck at details.  They just end with no detail. 

I thought the MCU was getting too hot so I did an upgrade for the MCU case and put in a fan.  I thought that worked but today after many prints without problems it crashed in the last quarter of a 3 hour print. 

I’ve replaced the USB cable three times and tried all the suggestions people have put out. 

I am ready to just go back to Marlin and check out the high speed firmware and forks. 

Has anyone else had this problem and resolved it?

**Comment by bassamanator** [comment]
How were things on marlin previously?

**Reply by klb331** [reply]
No problems at all. 

I’m seeing post after post after post about power issues. When I set up my pie, I was careful to use a USB cord that I put a piece of electrical tape over the power pin. I just checked it and I had unplugged and plugged the cable back in a few times, making sure that it was seated and I think I might’ve dislodged the electrical type. I’m gonna give it another day or so and see if that’s resolved the problem.

**Comment by bassamanator** [comment]
What errors, if any, are reported in the klipper console when the crashes occur? Is the raspberry pi useable after the crashes?

**Reply by klb331** [reply]
None. Klipper Shutdown. 

Yeah, I can use the Klipper touchscreen to reboot it and I can SSH into it.

**Reply by bassamanator** [reply]
So you're using the sovol klipper screen? I would assume it's defective, perhaps get a replacement.

If you have a spare computer of any kind, you can (temporarily) convert it to serve as a klipper host (instead of the sovol screen), and see how things go.

**Reply by klb331** [reply]
No. I am using a raspberry pie 3B.  It was new (I bought it a while ago but never used it) out of the box and I’m using an official power supply for it. 

As I said in my last post, I taped down the power for the USB connection and tested to make sure that it was blocking any power.  I hoped this would solve the problem…. But this morning it locked up an hour into a 3 hour print. 

I’ve regularly checked for updates to Klipper and applied updates and updated the firmware as well 

I tried this morning lowering the baud rate on the serial connection but when I did that Klipper couldn’t connect to the printer so I had to go back to the default number 250000

I’m burned out trying.  Ive wasted so much time and filament trying to get this to work. 

I’m going to wipe the drive and try from scratch again since I’ve added some macros like CV’s settings and macros and see what happens but if this happens again, I’m going back to Marlin.

**Reply by klb331** [reply]
In answer to your earlier question, this morning when it locked up I tested and yes, I could get into the Pi, I could get into the Klipper screen interface and reboot both but till I restarted the printer I couldn’t do anything.

**Reply by bassamanator** [reply]
You've brought up `power` and `USB cables` several times so let me respond to that. If you followed the link on my repo that shows you how to block the power pin on USB cables, here's a way to test out whether you did it **correctly**:
1. The printer is `off`.
2. You connect the Rpi to the printer motherboard.
3. The Z probe light should not turn on, nor should the screen for the SV06.
4. You are able to communicate with the printer via the Klipper dashboard (mainsail/fluidd) after you turn the printer `on`.

When you start your printer, you need to click `firmware restart` to 'start'.
![image](https://github.com/user-attachments/assets/1c93497e-a40b-492e-9863-985d813dc5ed)
After you do that, you should see no errors being reported, if you are, then `your Klipper is not good to go`.

You need not compile your own firmware and mess with those settings, because I compiled the firmware myself, and it is `tried, tested, and true`.

> I’m going to wipe the drive and try from scratch again since I’ve added some macros like CV’s settings and macros

So you're new to Klipper, and you decided to mess with the most popular Klipper configuration and how-to for these printers? Makes perfect sense.

**Reply by klb331** [reply]
So I should not update the firmware on my SV06 even if a new version of Klipper comes out? 

At this point I install the Mainsail image from the Raspberry Pi creator and then I install kiauh. I believe the Raspberry Pi builder installs the latest Klipper version  Should I not update anything?

I'll follow your page again with your config and we'll see how it goes --

**Comment by torayn** [comment]
Hey klb331!
I'm sorry to hear you're having problems, just want to give some of my experience in case applicable. How long have you had your SV06? By chance, is it in an enclosed space? Has the build-plate's power wire been hitting anything? I ask that because your issue is precisely what happened to my printer previously, it would be running just fine, and inexplicably, boop! 12 hour print interrupted only several hours in! So to be more clear, it was my build plates power cable that had gotten degraded through friction or other forces and I had to replace it, I used a 14 AWG cable to solder an replace it and wrapped it with some automotive tape. I have attached photo's of what I mean. I recommend checking the connections on your mobo of the printer and the ones on the build plate, and be very careful with all the screws  they're easy to strip. I hope you have a simple issue such as this, and not a more complex one at a firmware or software level, just want to put this out there in case it helps. 
![IMG_2964](https://github.com/user-attachments/assets/9872ced4-74fc-472c-b28c-bc8633bcfd14)
![IMG_2965](https://github.com/user-attachments/assets/4f6cccea-bcf8-4d35-aaee-21c35bf77c0b)
